data-plane/dispatcher-vertx/src/main/java/dev/knative/eventing/kafka/broker/dispatchervertx/VertxKafkaConsumer.java

```diff
@@ -15,8 +15,104 @@
  */
 package dev.knative.eventing.kafka.broker.dispatchervertx;
 
-public class VertxKafkaConsumer {
-    public static void main(String[] args) {
-        System.out.println("Hello Vertx KafkaConsumer!");
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import dev.knative.eventing.kafka.broker.dispatcher.ReactiveKafkaConsumer;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.kafka.client.common.KafkaClientOptions;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+
+public class VertxKafkaConsumer<K, V> implements ReactiveKafkaConsumer<K, V> {
+
+    private KafkaConsumer<K, V> consumer;
+
+    public VertxKafkaConsumer(Vertx v, KafkaClientOptions configs) {
+        consumer = KafkaConsumer.create(v, configs);
+    }
+
+    @Override
+    public Future<Map<TopicPartition,OffsetAndMetadata>> commit(Map<TopicPartition,OffsetAndMetadata> offset){
+
+        Map<io.vertx.kafka.client.common.TopicPartition,io.vertx.kafka.client.consumer.OffsetAndMetadata> vertxOffset = new HashMap<>();
+
+        for(Map.Entry<TopicPartition,OffsetAndMetadata> entry: offset.entrySet()){
+            vertxOffset.put(
+                new io.vertx.kafka.client.common.TopicPartition(entry.getKey().topic(), entry.getKey().partition()),
+                new io.vertx.kafka.client.consumer.OffsetAndMetadata(entry.getValue().offset(), entry.getValue().metadata())
+            );
+        }
+        
+        return consumer.commit(vertxOffset).map( v -> {
+            Map<TopicPartition,OffsetAndMetadata> result = new HashMap<>();
+            for(Map.Entry<io.vertx.kafka.client.common.TopicPartition,io.vertx.kafka.client.consumer.OffsetAndMetadata> entry: v.entrySet()){
+                result.put(
+                    new TopicPartition(entry.getKey().getTopic(), entry.getKey().getPartition()),
+                    new OffsetAndMetadata(entry.getValue().getOffset(), entry.getValue().getMetadata())
+                );
+            }
+            return result;
+        });
```

Comment:
For both `for loop`s before commit and inside `map`, can we use the Stream API?

For example:
```java
final var vertxOffset = offset.entrySet().stream().
  map(entry -> new SimpleImmutableEntry(
      new io.vertx.kafka.client.common.TopicPartition(entry.getKey().topic(), entry.getKey().partition()),
      new io.vertx.kafka.client.consumer.OffsetAndMetadata(entry.getValue().offset(), entry.getValue().metadata())
  )).
  collect(Collectors.toMap(Entry::getKey, Entry::getValue))
```